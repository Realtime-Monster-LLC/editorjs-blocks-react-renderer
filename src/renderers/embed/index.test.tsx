import React from 'react';
import { create } from 'react-test-renderer';
import Embed, { EmbedBlockData } from '.';

describe('<Embed />', () => {
  describe('when receives a Embed block with actions', () => {
    const data: EmbedBlockData = {
      service: 'youtube',
      source: 'https://www.youtube.com/watch?v=yDiD8F9ItX0',
      embed: 'https://www.youtube.com/embed/yDiD8F9ItX0',
      width: 580,
      height: 320,
      caption: 'Deep in the Universe',
    };

    it('renders a <figure> block with <iframe> and <figcaption>', () => {
      expect(create(<Embed data={data} className="special-embed" />).toJSON()).toMatchSnapshot();
    });
  });
});
