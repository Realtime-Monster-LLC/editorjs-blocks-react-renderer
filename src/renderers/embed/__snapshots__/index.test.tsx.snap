// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Embed /> when receives a Embed block with actions renders a <figure> block with <iframe> and <figcaption> 1`] = `
<figure>
  <iframe
    className="special-embed embed-block-service-youtube"
    data-src="https://www.youtube.com/watch?v=yDiD8F9ItX0"
    frameBorder="0"
    height="320"
    src="https://www.youtube.com/embed/yDiD8F9ItX0"
    width="580"
  />
  <figcaption>
    Deep in the Universe
  </figcaption>
</figure>
`;
